internal/common/teleport.go

```diff
@@ -0,0 +1,62 @@
+package common
+
+import (
+	"context"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/giantswarm/cluster-test-suites/internal/helper"
+	"github.com/giantswarm/cluster-test-suites/internal/state"
+	"github.com/giantswarm/clustertest/pkg/logger"
+	"github.com/giantswarm/clustertest/pkg/wait"
+
+	tc "github.com/gravitational/teleport/api/client"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func runTeleport(teleportSupported bool) {
+	Context("teleport", func() {
+		var teleportClient *tc.Client
+
+		isClusterRegistered := func(clusterName string) (bool, error) {
+			clusters, err := teleportClient.GetKubernetesServers(context.Background())
```

Comment:
turns out `GetKubernetesCluster` doesn't return any registered cluster, so had to use `GetKubernetesServers`